client/utils/currency/index.js

```diff
@@ -60,7 +60,7 @@ export const getCurrency = ( currencyCode, baseCurrencyCode = null ) => {
 				currency.decimalSeparator = baseCurrency.decimalSeparator;
 				currency.thousandSeparator = baseCurrency.thousandSeparator;
 				currency.symbolPosition = baseCurrency.symbolPosition;
-				if ( currency.precision !== 0 ) {
+				if ( currency.precision !== 0 && country !== 'JP' ) {
```

Comment:
I certainly see many problems from that fix:

1. Country and currency are not the same things. 
2. This logic doesn't explain why only JP is exception.
3. The fix is for very particular case. What about any other zero-decimal currencies?

Overall, from the quick glance it feels like the fix is for the symptom, not for the root problem.

Reply:
I agree. This is a temporary fix until we explore if removing the precision overwriting would cause any issues. 

Initially, I tried to rely on the `baseCurrencyCode`, but it is not always present, and we cannot rely on it.

Reply:
I also added a small comment stating that this is a temporary fix https://github.com/Automattic/woocommerce-payments/pull/6945/commits/3d557429c22015107cb36a0f190161f381e2d660

Reply:
Thanks, with comment it looks much better.